src/pd/async/util.rs

```diff
@@ -74,34 +82,52 @@ impl LeaderClient {
     }
 
     // Re-establish connection with PD leader in synchronized fashion.
-    pub fn reconnect(&self) {
-        let start = Instant::now();
-        let ret = try_connect_leader(&self.inner.rl().members);
+    pub fn reconnect(&self) -> PdFuture<()> {
```

Comment:
But it's not. When a sleep is returned then completed, just mean time passes and nothing else happens.